mods/mapgen/decorations.lua

```diff
@@ -0,0 +1,758 @@
+	decoration = "blocks:marble_mese_circuits_under",
+})
+
+minetest.register_decoration({
+	name = "mese_tech_crystal",
+	deco_type = "simple",
+	place_on = {"blocks:stone_mese_circuits", "blocks:marble_mese_circuits"},
+	sidelen = 12,
+	fill_ratio = 0.05,
+	y_max = mapgen.underground_limit,
+	y_min = mapgen.world_bottom,
+	flags = "all_floors",
+	height = 3,
+	height_max = 6,
+	decoration = "blocks:mese_tech_crystal",
+})
+
+minetest.register_decoration({
+	name = "mese_vines",
+	deco_type = "simple",
+	place_on = {"blocks:stone_mese_circuits_under", "blocks:marble_mese_circuits_under"},
+	noise_params = {
+		offset = -0.25,
+		scale = 1,
+		spread = {x = 4, y = 4, z = 4},
+		seed = 62262226,
+		octaves = 1,
+		flags = "eased",
+	},
+	y_max = mapgen.underground_limit,
+	y_min = mapgen.world_bottom,
+	flags = "all_ceilings",
+	height = 3,
+	height_max = 8,
+	decoration = "flora:vines_mese",
+})
+
+minetest.register_decoration({
+	name = "mese_moss",
+	deco_type = "simple",
+	place_on = {"blocks:stone_mese_circuits_under", "blocks:marble_mese_circuits_under"},
+	noise_params = {
+		offset = -0.25,
+		scale = 2,
+		spread = {x = 4, y = 4, z = 4},
+		seed = 262226222,
+		octaves = 1,
+		flags = "absvalue, eased",
+	},
+	y_max = mapgen.underground_limit,
+	y_min = mapgen.world_bottom,
+	flags = "all_ceilings",
+	decoration = "flora:moss_mese",
+})
+
+for i=1,5 do
+	-- Small plants
+	minetest.register_decoration({
+		name = "mese_flower_" .. i,
+		deco_type = "simple",
+		place_on = {"blocks:stone_mese_circuits", "blocks:marble_mese_circuits"},
+		fill_ratio = 0.1/5,
+		y_max = mapgen.underground_limit,
+		y_min = mapgen.world_bottom,
+		param2 = 1,
+		flags = "all_floors",
+		decoration = "flora:flower_mese_" .. i,
+	})
+end
+
+for i=1,5 do
+	-- Grass
+	minetest.register_decoration({
+		name = "mese_grass_" .. i,
+		deco_type = "simple",
+		place_on = {"blocks:stone_mese_circuits", "blocks:marble_mese_circuits"},
+		fill_ratio = 1/5,
+		y_max = mapgen.underground_limit,
+		y_min = mapgen.world_bottom,
+		param2 = 1,
+		flags = "all_floors",
+		decoration = "flora:grass_mese_" .. i,
+	})
+end
+
+-- Other Decorations
+
+minetest.register_decoration({
+	name = "sfcave_mossycobble",
+	deco_type = "simple",
+	place_on = {"blocks:stone"},
+	fill_ratio = 10.0,
+	y_max = 0,
+	y_min = mapgen.underground_start,
+	spawn_by = "blocks:dirt_with_grass",
+	num_spawn_by = 1,
+	flags = "all_floors, all_ceilings, force_placement",
+	place_offset_y = -1, -- Requires force_placement
+	decoration = "blocks:mossycobble",
+})
+
+minetest.register_decoration({
+	name = "tree_vines",
+	deco_type = "simple",
+	place_on = {"flora:leaves"},
+	fill_ratio = 0.2,
+	y_max = 31000,
+	y_min = mapgen.underground_start,
+	flags = "all_ceilings",
+	height = 3,
+	height_max = 8,
+	decoration = "flora:vines",
+})
+
+minetest.register_decoration({
+	name = "quests:spaceship",
+	deco_type = "schematic",
+	place_on = {"blocks:sand"},
+	sidelen = 16,
+	fill_ratio = 0.00000001,
```

Comment:
I think cut this back by another couple of zeros. Either i've been really lucky or, i've consistently been managing to find one within a couple of minutes.

Reply:
It doesn't help, I didn't notice any improvement after adding more and more zeroes, it seems it always just wants to place at least one ship per a few chunks. Not sure how it works honestly

Reply:
Ah, that's annoying, there should probably be some dedicated way to place them like nyancats then.
I'll make this as resolved for now, and I think it can be revisited again as I convert quest into a more solid framework.